plone/app/event/portlets/portlet_events.py

```diff
@@ -27,6 +28,46 @@
 from zope.interface import implementer
 
 
+def _render_events_cachekey(method, self):
+    # XXX: Since the events portlet shows the upcoming events, I will simply get
+    # the most recently modified event using the same logic, and return its uuid
+    # and last modified date as key. If there is a change, then the most expensive
+    # computation can be performed
+
+    site = getSite()
+    pc = site.portal_catalog
+
+    query = {}
+    data = self.data
+    if data.state:
+        query['review_state'] = data.state
+
+    query.update(self.request.get('contentFilter', {}))
+    if ICollection and ICollection.providedBy(self.search_base):
```

Comment:
Sorry, there's a missing import.

Reply:
```suggestion
    if ISyndicatableCollection and ISyndicatableCollection.providedBy(self.search_base):
```
This got changed on master ...